provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/task/ProvisioningTask.java

```diff
@@ -247,19 +243,27 @@ private void executeStateMachineForStatus(RegistrationOperationStatusParser regi
                     break;
                 case FAILED:
                     this.dpsStatus = PROVISIONING_DEVICE_STATUS_FAILED;
-                    String errorMessage = statusRegistrationOperationStatusParser.getRegistrationState().getErrorMessage();
+                    String errorMessage = registrationOperationStatusParser.getRegistrationState().getErrorMessage();
                     ProvisioningDeviceHubException dpsHubException = new ProvisioningDeviceHubException(errorMessage);
-                    dpsHubException.setErrorCode(registrationOperationStatusParser.getRegistrationState().getErrorCode());
+                    if (registrationOperationStatusParser.getRegistrationState().getErrorCode() != null)
```

Comment:
This null check is the relevant change that fixes the bug here. Going from Integer to int we weren't checking for null